src/error.ts

```diff
@@ -0,0 +1,26 @@
+abstract class SeedError extends Error {
+  constructor(
+    message: string,
+    public readonly code: string,
+    public readonly context?: any
+  ) {
+    super(message);
+    this.name = this.constructor.name;
+
+    if (Error.captureStackTrace) {
+      Error.captureStackTrace(this, this.constructor);
+    }
+  }
+}
+
+export class PreAuthorizationFailedError extends SeedError {
```

Comment:
Do we need `Error` postfix?
cf) Boom
<img width="585" alt="image" src="https://github.com/user-attachments/assets/394b4904-25fe-4aaf-a586-dd0b086a03bd">